plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1787,6 +1809,25 @@ private void checkUnsupportedGeneratedColumns(MetadataEntry metadataEntry)
         }
     }
 
+    private void checkUnsupportedColumnMappingId(MetadataEntry metadataEntry)
+    {
+        ColumnMappingMode columnMappingMode = getColumnMappingMode(metadataEntry);
+        if (columnMappingMode == ColumnMappingMode.ID) {
```

Comment:
What about flipping the check ?

```
if (!(columnMappingMode == NONE || columnMappingMode == ColumnMappingMode.NAME))
```

In the (unlikely) event a different column mapping is introduced, we'd catch this situation here.
```

Reply:
done